core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1699,7 +1699,8 @@ void completeFileInternal(RpcContext rpcContext, LockedInodePath inodePath,
       try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
         UnderFileSystem ufs = ufsResource.get();
         if (ufsStatus == null) {
-          ufsFingerprint = ufs.getParsedFingerprint(ufsPath).serialize();
+          ufsFingerprint = ufs.getParsedFingerprint(ufsPath, context.getOptions().hasContentHash()
+              ? context.getOptions().getContentHash() : null).serialize();
```

Comment:
will we have a case where input contentHash is diff than the ufs status contenthash, meaning at time of completefile ufs has changed to some other content, and we are updating fileinode with our version of contenthash but with other field uptodate with ufs like owner ?  or at least we could reload if we find contenthash mismatch later

Reply:
What I am going for here is to have the data in Alluxio match the fingerprint created. The first part is to match the contents of the file which is done here. The next part will be to match the metadata (this PR is still in progress). I might be able to remove the UFS access here all together.

I am thinking that even if the data in Alluxio is out of sync with the UFS it is fine here, as later synchronizations will happen based on the metadata sync policy. The main thing that matters is that the fingerprint matches the actual data so that if there is an updated file on the UFS it will be synced whenever the next sync happens based on the policy.

Reply:
So after internal discussion, only the content hash of the file will be kept consistent.

This is due to other metadata in the fingerprint being computed in so many different ways, with all kinds of strange behaviors that clients might rely on, it was better not to mess with this as it could break things.